schema/snapcraft.yaml

```diff
@@ -202,6 +202,27 @@ properties:
             description: What kind of wrapper to generate for the given command
             enum:
               - none
+          sockets:
+            type: object
+            additionalProperties: false
```

Comment:
Now that #1615 has landed, I have another suggestion. When the schema is:

1. An object
2. With no additional properties allowed
3. That uses `patternProperties` rather than hard-coded properties

... and is handed an object with properties that do not match the pattern, rather than barfing on the pattern, jsonschema unhelpfully barfs on the `additionalProperties: false` instead. So a YAML like this:

```
apps:
  foo:
    command: echo "hello"
    sockets:
      BAD_SOCKET:
        listen-stream: bar
```
Results in this:
```
Issues while validating snapcraft.yaml: The 'apps/foo/sockets' property does not match the required schema: Additional properties are not allowed ('BAD_SOCKET' was unexpected)
```
That can be improved if you add a `validation-failure` message like this:
```
sockets:
  type: object
  additionalProperties: false
  validation-failure:
    "{!r} is not a valid socket name. Socket names consist of
    lower-case alphanumeric characters and hyphens."
  patternProperties:
    "^[a-z0-9][a-z0-9+-]*$":
      <snip>
```
Then the error message becomes:
```
Issues while validating snapcraft.yaml: The 'apps/foo/sockets' property does not match the required schema: 'BAD_SOCKET' is not a valid socket name. Socket names consist of lower-case alphanumeric characters and hyphens.
```

Reply:
updated schema as suggested